whelan-low

whelan-low opened the pull request:
# **Greg the Assistant**

> "Your mind is for having ideas, not holding them" - David Allen

A chat-bot to free your brain from remembering your tasks! It can:
- Help you find your tasks easily!
- Store as many things as you need
- Supports many types of tasks!
- Extremely simple to use!

All you need to do is:
1. Download it from [here](github.com/whelan-low/ip)
2. double-click it
3. add your tasks
4. Let Greg handle the rest!

It is completely **FREE!!**

Features:
- [x] Task management
- [ ] Deadline management (Coming soon!)
- [ ] Event management (Coming soon!)

If you are a Java programmer, you can use it to practice Java too.
Here is the main method:
`     public static void main(String[] args) {
        Ui.startup();
    } `

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

whelan-low committed:
Level 0

whelan-low committed:
Level 1

whelan-low committed:
Level 2

whelan-low committed:
Level 3

whelan-low committed:
Level 4

whelan-low committed:
A-TextUiTesting

whelan-low committed:
Level 5

whelan-low committed:
Level 6

whelan-low committed:
Level 7

whelan-low committed:
Level 8

whelan-low committed:
A-MoreOOP

whelan-low committed:
A-Packages

whelan-low committed:
A-Gradle

whelan-low committed:
no message

whelan-low committed:
Merge commit 'f837ddb7d2b77f6654758f93d056420a580787ca'

whelan-low committed:
A-Gradle

whelan-low committed:
A-JUnit

whelan-low committed:
Merge remote-tracking branch 'origin/add-gradle-support'

whelan-low committed:
adding branch

whelan-low committed:
gradle update

whelan-low committed:
Merge branch 'branch-Level-7'

whelan-low committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	.gitignore
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

whelan-low committed:
A-Jar

whelan-low committed:
removed unused import

whelan-low committed:
A-JavaDoc

whelan-low committed:
A-CheckStyle

whelan-low committed:
Level 9

whelan-low committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

# Conflicts:
#	src/main/java/duke/Task.java

whelan-low committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

whelan-low committed:
Level 9

whelan-low committed:
Merge files fixed

kinjalagarwal1810 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,265 @@
+     */
+    public Task get(int index) {
+        return tasks.get(index);
+    }
```
Perpahs the name of the variable could be more specific like "getTask" so it is easily understandable

kinjalagarwal1810 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,265 @@
+     *
+     * @return the string representation of the task list.
+     */
+    public String list() {
```
You can consider naming your method here more specifically like "printList" so we know it is printing or returning a string.

kinjalagarwal1810 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,265 @@
+     *
+     * @param task the task being added
+     */
+    public void announceAddition(Task task) {
```
Perhaps instead of annouce you can use print since it is more commonly used when we return a string

kinjalagarwal1810 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,77 @@
+     * Runs the shutdown sequence.
+     * Saves then says bye.
+     */
+    public void bye() {
```
The name of method would be better if it was a verb so maybe use "printBye" or another verb.

kinjalagarwal1810 reviewed (commented):
Overall looks very well done! I've just added a couple of minor comments.

whelan-low committed:
Level 10

whelan-low committed:
Created new JAR file

whelan-low committed:
Handled errors that can happen when parsing dates.

This should reduce the chance of commands not working
after trying to put an invalid date when creating a deadline or event task.

whelan-low committed:
Assertions added.

Helps to reduce the chance of an invalid input

whelan-low committed:
CodeQuality

whelan-low committed:
Merge branch 'branch-A-CodeQuality'

whelan-low committed:
Added JavaDoc

whelan-low committed:
B-ViewSchedules

whelan-low committed:
added commit

whelan-low committed:
Revert "added commit"

This reverts commit 6a3ad2da065a02700fcf8a7d8dd8b0d961714449.

whelan-low committed:
Merge pull request #1 from whelan-low/Merging-PRs

Merging branches

whelan-low committed:
A-MoreErrorHandling

whelan-low committed:
Merge pull request #3 from whelan-low/Merging-PRs

Adding to master

whelan-low committed:
Fixed more boolean naming

whelan-low committed:
Merge branch 'Merging-PRs'

whelan-low committed:
Changed the build.gradle file

whelan-low committed:
Added product image

whelan-low committed:
User Guide added

whelan-low committed:
Updated user guide again

whelan-low committed:
Updated user guide - 3rd update

whelan-low committed:
Updated user guide - Update 4

whelan-low committed:
updated user guide - Update 5

whelan-low committed:
User guide update 6

whelan-low committed:
Add Test Cases

whelan-low committed:
Add Javadocs

whelan-low committed:
Add more Javadocs

whelan-low committed:
Add Launcher class Javadoc

whelan-low committed:
Add Javadocs for the Help class
